framework/source/class/qx/ui/window/Window.js

```diff
@@ -1050,5 +1210,29 @@ qx.Class.define("qx.ui.window.Window",
       this.close();
       this.getChildControl("close-button").reset();
     }
+  },
+
+  destruct : function()
+  {
+    var id;
+    var parent;
+
+    // Remove ourselves from the focus handler
+    qx.ui.core.FocusHandler.getInstance().removeRoot(this);
+
+    // Remove the listener for appear, if there is one
+    id = this.__centeringAppearId;
+    id && this.removeListenerById(id);
```

Comment:
That two lines can be omitted since listeners to this will be automatically removed.